Core/clim-basic/extended-streams/stream-output.lisp

```diff
@@ -328,38 +328,39 @@
-                          (text-style-width text-style medium)
-                          (lambda (string start end)
-                            (text-size medium string
-                                       :text-style text-style
-                                       :start start :end end))))
-               (splits (line-breaks string width
-                                    :initial-offset (- cx left-margin)
-                                    :margin (- right-margin left-margin)
-                                    :break-strategy (ecase eol-action
-                                                      (:wrap NIL)
-                                                      (:wrap* T))
-                                    :start start :end end)))
-          (do ((start start (car split))
-               (split splits (rest split)))
-              ((null split)
-               (stream-write-output stream string start end)
-               (setf (cursor-position (stream-text-cursor stream))
-                     (values (+ left-margin
-                                (stream-string-width stream string
-                                                     :start start :end split
-                                                     :text-style text-style))
-                             (nth-value 1 (stream-cursor-position stream)))))
-            (ecase eol-action
-              (:wrap  (stream-write-output stream string start (car split)))
-              (:wrap* (let ((pos (position #\space string
-                                           :from-end t :start start :end (car split)
-                                           :test-not #'char=)))
-                        (when pos (incf pos))
-                        (stream-write-output stream string start (or pos (car split))))))
-            ;; print a soft newline
-            (seos-write-newline stream t)))))))
+        ;;
+        ;; Writing a newline may cause the cursor increment, so we
+        ;; need to compute split for each line after the soft newline
+        ;; has been written. -- jd 2020-03-01
+        (loop with width = (if (text-style-fixed-width-p text-style medium)
+                               (text-style-width text-style medium)
+                               (lambda (string start end)
+                                 (text-size medium string
+                                            :text-style text-style
+                                            :start start :end end)))
+              with margin = (- right-margin left-margin)
+              with cursor = (stream-text-cursor stream)
+              with break  = (ecase eol-action
+                              (:wrap NIL)
+                              (:wrap* T))
+              for offset = (- (cursor-position cursor) left-margin)
+              for split  = (car (line-breaks string width
+                                             :count 1
```

Comment:
Is the increased cost of doing this repeatedly somewhat compensated by the `:count 1`?

Reply:
it is, that was one of reasons behind rewriting break-lines utility.